internal/resource/route.go

```diff
@@ -663,6 +728,53 @@ func init() {
 					},
 				},
 			},
+			{
+				Title:       ExpressionRouteRuleTitle,
+				Description: "When 'expression' is defined, 'priority' is required.",
+				If: &generator.Schema{
+					Required: []string{"expression"},
+				},
+				Then: &generator.Schema{
+					Required: []string{"priority"},
+				},
+			},
+			{
+				Title: ExpressionRouteRuleTitle,
+				If: &generator.Schema{
+					Required: []string{"expression"},
+				},
+				Then: &generator.Schema{
+					Properties: map[string]*generator.Schema{
+						"hosts": {Not: &generator.Schema{
+							Description: "When 'expression' is defined, the field 'hosts' must not be set.",
```

Comment:
Can we please generate this string programmatically to avoid the repetition?

Reply:
done